consent-ws/src/main/java/org/broadinstitute/consent/http/resources/DataUseLetterResource.java

```diff
@@ -125,13 +126,17 @@ public Consent updateDUL(
     @GET
     @Produces(MediaType.APPLICATION_OCTET_STREAM)
     @RolesAllowed({"ADMIN","CHAIRPERSON","MEMBER","DATAOWNER"})
-    public Response getDUL(@PathParam("id") String consentId) {
-        String msg = String.format("GETing Data Use Letter for consent with id '%s'", consentId);
+    public Response getDUL(@PathParam("id") String consentId, @QueryParam("electionId") Integer electionId) {
+        String msg = String.format("GETing Data Use Letter for consent with id '%s' and Election Id '%s", consentId, electionId);
         logger().debug(msg);
+        Election election = null;
         try {
             Consent consent = api.retrieve(consentId);
-            String fileUrl  = consent.getDataUseLetter();
-            String fileName = consent.getDulName();
+            if (StringUtils.isNotEmpty(consent.getLastElectionStatus())) {
```

Comment:
You could check if electionId is null and if so retrieve the consent and get the the url and name from the consent. Otherwise call retrieveElectionDul and get the url and name from the election. In this second case if the election is null, you could throw an UnknownIdentifierException for the election id.

Reply:
I see what you mean, it sounds good, but that way we would lose the ability to bring
a dul from the last election version without passing an electionId, if such exists (which is used from the ui).

The reason why is because:

first we bring consent, the method used here brings some fields from its most recent election if this exists. If it doesnt the value in LastElectionStatus is null.

So, if consent.lastElectionStatus is null then it means that there's no election associated to
the consent for which we use consent's dul and dulurl.

On the other hand, if this value is not null,  it proceeds to check if an electionId is passed, resulting to bring the last election when the value is null, or the corresponding to the passed id.

You can see this last behavior described above, in retrieveElection, where we also throw the exception you mentioned.

Please if you agree to make this change anyway let me know.